tests/setup.js

```diff
@@ -13,6 +15,10 @@ import 'jest-enzyme';
 
 import 'amo/polyfill';
 
+// These are needed for Jest + jsdom v17.
+global.TextEncoder = TextEncoder;
+global.TextDecoder = TextDecoder;
```

Comment:
This is described in https://github.com/jsdom/jsdom/issues/2524#issuecomment-902027138 as:

> [...] you can polyfill TextEncoder and TextDecoder, but loading jsdom within jsdom doesn't seem like something that's supposed to happen

Now the question is: why are we loading jsdom within jsdom?

Reply:
And follow-up question: should we try to fix our jest config instead? (`testEnvironment`)

Reply:
> Now the question is: why are we loading jsdom within jsdom?

Let me see if that is in fact what we're doing here (as opposed to Jest doing it without our involvement). If so then maybe that comment gives us a clue about what we can do about it.

Reply:
It seems like we *are* in fact loading `jsdom` within `jsdom` when we use the code [here](https://github.com/mozilla/addons-frontend/blob/master/src/amo/window.js#L1).

I assume we need this `amo/window.js` to make the DOM available on node, for SSR. I can remove this code entirely and all the tests pass, and `jsdom` no longer throws the exception being discussed here, but it breaks SSR. 

So we need this code when running the server, but we don't actually need the code when tests are running. I don't think we want to change our production code to be aware of whether it's being run in a test or not, so I'm not sure what to do about this, other than use the band-aid fix that seems to work with the current patch.

Reply:
> And follow-up question: should we try to fix our jest config instead? (testEnvironment)

@willdurand I tried this, and have attached the attempt in a commit, but I couldn't seem to get it working. I created a custom test environment, which was being picked up (I verified by adding some console statements), but it doesn't seem to allow me to set the globals in a way that will work for jsdom.

As you can see from the new commit, I tried adding the globals in the `setup` method, both via `globals` and `this.globals` (the latter of which is [documented in the docs](https://jestjs.io/docs/configuration#testenvironment-string)). I think the problem is that this makes the variables available as globals *for the test*, but what we really need is for them to be available as globals *to jsdom*, and I haven't been able to figure out how to do that via `testEnvironment`. Perhaps you have an idea about this.

TBPH, I'm not sure I see the value in having this defined in a custom test environment as opposed to in `tests/setup.js`. The latter seems just as appropriate to me, and perhaps is the only way to make it work?